src/actions/actions/services/dispatchers/verusid/verusid.js

```diff
@@ -56,3 +69,212 @@ export const unlinkVerusId = async (iAddress, chain) => {
+    {
+      ...serviceData,
+      pending_ids: currentPendingIdentities,
+    },
+    VERUSID_SERVICE_ID,
+    state.authentication.activeAccount.accountHash,
+  );
+};
+
+export const deleteAllProvisionedIds = async () => {
+  const state = store.getState();
+
+  if (state.authentication.activeAccount == null) {
+    throw new Error('You must be signed in for ID provisioning');
+  }
+
+  const serviceData = await requestServiceStoredData(VERUSID_SERVICE_ID);
+
+  return await modifyServiceStoredDataForUser(
+    {
+      ...serviceData,
+      pending_ids: {},
+    },
+    VERUSID_SERVICE_ID,
+    state.authentication.activeAccount.accountHash,
+  );
+};
+
+export const checkVerusIdNotificationsForUpdates = async () => {
+  const state = store.getState();
+
+  const getPotentialPrimaryAddresses = async (coinObj, channel) => {
+    const seeds = await requestSeeds();
+
+    const seed = seeds[channel];
+
+    const keyObj = await deriveKeyPair(seed, coinObj, channel);
+    const {addresses} = keyObj;
+
+    return addresses;
+  };
+
+  if (state.authentication.activeAccount == null) {
+    throw new Error('You must be signed in for ID provisioning functions');
+  }
+  const isTestnet = Object.keys(state.authentication.activeAccount.testnetOverrides).length > 0;
+
+  const system = isTestnet ? coinsList.VRSCTEST : coinsList.VRSC;
+  const ticker = system.id;
+
+  // Itterate through all pending IDs and check for updates
+  const pendingIds = state.channelStore_verusid.pendingIds;
+
+  const serviceData = await requestServiceStoredData(VERUSID_SERVICE_ID);
+  const currentLinkedIdentities =  Object.keys(serviceData.linked_ids && serviceData.linked_ids[ticker] || {});
+
+  if (pendingIds[ticker]) {
+    const details = Object.keys(pendingIds[ticker]);
+    for (const iaddress of details) {
+
+      // once an ID is linked, remove it from pending IDs, or if the server has rejected it delete.
+      if (pendingIds[ticker][iaddress].status === NOTIFICATION_TYPE_VERUSID_READY) {
+        if (currentLinkedIdentities.indexOf(iaddress) > -1 || pendingIds[ticker][iaddress].status === NOTIFICATION_TYPE_VERUSID_FAILED) {
+          await deleteProvisionedIds(iaddress, ticker);
+          await updatePendingVerusIds();
+          await dispatchRemoveNotification(pendingIds[ticker][iaddress].notificationUid);
+        }
+        continue;
+      } 
+        
+        if ((pendingIds[ticker][iaddress].createdAt + 600) < Math.floor(Date.now() / 1000) &&
+              pendingIds[ticker][iaddress].status !== NOTIFICATION_TYPE_VERUSID_READY) {
+          // If the request is older than 10 minutes, check info endpoint to see if it was accepted or rejected
+        let errorFound = false;
+        try {
+          if (pendingIds[ticker][iaddress].infoUri) {
+            const response = await axios.get(pendingIds[ticker][iaddress].infoUri);
+            const responseData = new primitives.LoginConsentProvisioningDecision(response.data);
```

Comment:
The server needs to return a signed LoginConsentProvisioningResponse, which the app then needs to verify, and ensure that the ID it is signed by matches the ID of the original loginconsentrequest. Refer to src/\_\_tests\_\_/vdxf/loginconsent.test.ts to see how the LoginConsentProvisioningResponse is used.